CRM/Member/Form/MembershipRenewal.php

```diff
@@ -598,6 +598,10 @@ public function postProcess() {
 
       $payment = CRM_Core_Payment::singleton($this->_mode, $this->_paymentProcessor, $this);
 
+      if ($paymentParams['auto_renew']) {
+        $contributionRecurParams = $this->processRecurringContribution($paymentParams);
```

Comment:
@eileenmcnaughton : function call to processRecurringContribution should be done somewhere down after we have checked if $result was erroneous or not.